privilege/privileges/cache.go

```diff
@@ -50,10 +50,11 @@ func computePrivMask(privs []mysql.PrivilegeType) mysql.PrivilegeType {
 
 // UserRecord is used to represent a user record in privilege cache.
 type UserRecord struct {
-	Host       string // max length 60, primary key
-	User       string // max length 16, primary key
-	Password   string // max length 41
-	Privileges mysql.PrivilegeType
+	Host          string // max length 60, primary key
+	User          string // max length 16, primary key
+	Password      string // max length 41
```

Comment:
In MySQL 5.7 this was renamed `authentication_string` and changed to type `text` (so it can support other hash formats, like `caching_sha2_password`).  We will need to support `caching_sha2_password`, but I am a bit unsure how to implement this because users do run `select user,host,password from mysql.user`.  Should we create a virtual column for `authentication_string` to ensure compatibility?